test/gui/shared/steps/steps.py

```diff
@@ -511,3 +511,21 @@ def step(context, permissions, resource, password):
     clickButton(waitForObject(names.oCC_ShareLinkWidget_createShareButton_QPushButton))
     waitFor(lambda: (findObject(names.linkShares_0_0_QModelIndex).displayText == "Public link"))
 
+
+@When('the user creates a new public link for folder "|any|" with "|any|" using the client-UI')
+def step(context, resource, role):
+    resource = sanitizePath(substituteInLineCodes(context, resource))
+    radioObjectName = ''
+    
+
+    if role == 'Viewer':
+        radioObjectName = names.oCC_ShareLinkWidget_radio_readOnly_QRadioButton
+    elif role == 'Editor':
+        radioObjectName = names.oCC_ShareLinkWidget_radio_readWrite_QRadioButton
+    else:
```

Comment:
lets throw an exception is an unexpected role is given

Reply:
Done